File: framework/languages/java/e2e/RegistrationCodec/Server/Main/src/main/java/systems/zlink/e2e/registrationcodec/main/Configuration/ServerOptions.java
package systems.zlink.e2e.registrationcodec.main.Configuration;

public record ServerOptions(
    String serverEndpoint,
    String httpEndpoint,
    String logDir) {
    public static ServerOptions fromEnv() {
        return new ServerOptions(
            Env.get("ZLINK_JAVA_E2E_SERVER_ENDPOINT"),
            Env.get("ZLINK_JAVA_E2E_HTTP_ENDPOINT"),
            Env.get("ZLINK_JAVA_E2E_LOG_DIR", "logs"));
    }
}

File: framework/languages/java/e2e-kotlin/ResilienceLifecycle/Server/Registry/src/main/kotlin/systems/zlink/e2e/kotlin/resiliencelifecycle/Configuration/ServerOptions.kt
package systems.zlink.e2e.kotlin.resiliencelifecycle

data class ServerOptions(
    val registryPubEndpoint: String,
    val registryRouterEndpoint: String,
) {
    companion object {
        fun fromEnv(): ServerOptions =
            ServerOptions(
                registryPubEndpoint = Env.get("ZLINK_KOTLIN_E2E_REGISTRY_PUB"),
                registryRouterEndpoint = Env.get("ZLINK_KOTLIN_E2E_REGISTRY_ROUTER"),
            )
    }
}
